tests/test_infrastructure.py

```diff
@@ -0,0 +1,16 @@
+"""Tests of custom testing infrastructure"""
+
+import pytest
+import numpy as np
+from stcal.testing_helpers import MemoryThreshold, MemoryThresholdExceeded
+
+
+def test_memory_threshold():
+    with MemoryThreshold(1000):
```

Comment:
Looks like some conditions have too much overhead for this test and it's failing with python 3.12.
https://github.com/spacetelescope/stcal/actions/runs/11259184554/job/31307581409?pr=299

Reply:
thanks yeah noticed that